osu.Game/Updater/UpdateManager.cs

```diff
@@ -18,9 +18,11 @@ namespace osu.Game.Updater
     public class UpdateManager : CompositeDrawable
     {
         /// <summary>
-        /// Whether this UpdateManager is capable of checking for updates.
+        /// Whether this UpdateManager should be or is capable of checking for updates.
         /// </summary>
-        public virtual bool CanCheckForUpdate => false;
+        public bool CanCheckForUpdate => game.IsDeployedBuild;
```

Comment:
This will fail on iOS, maybe do:
```csharp
public bool CanPerformUpdate => game.IsDeployedBuild && CanCheckForUpdate;

protected virtual bool CanCheckForUpdate => false;
```
but not sure about the naming though, may want a more saner naming here

Reply:
Why would this be failing, because IsDeployedBuild is false on iOS?

Reply:
Yeah, I think as long as iOS reads the assembly version properly this should work with the simple update manager?

Reply:
iOS is on a completely different version publishing system (by TestFlight), hence the reason if you open `OsuGameIOS` you would find `SimpleUpdateManager` isn't used there.

With this staying as-is, `Check for update` button will be enabled for iOS, which we're likely not expecting here given how it's published.

Reply:
Fair. I got my virtual implementations confused.

Reply:
Fix in a slightly different way.